### ISSUE TITLE
SP5: Improve `Colorbar` layout and ticks

### ISSUE THREAD

**Issue by NateEbling** [ScottPlot/ScottPlot#3294]
**Question:** Can I add a color bar to a plot object without data and what needs to be passed in for the _source_ parameter? 

**ScottPlot Version:** 5.0.20

**Code Sample:** 

```cs
public Plot MakePlot()
{
    ScottPlot.Plot myPlot = new();
   
    myPlot.XLabel("X");
    myPlot.YLabel("Y");
    myPlot.Title("Title");
    
    myPlot.Add.ColorBar(   );

    return myPlot;
}
```

The source parameter is of type IHasColorAxis, but I'm not clear on what to pass in, or if this is even possible.

**Comment by swharden** [comment]
This is close but not working perfectly yet. 

```cs
double[,] data = SampleData.MonaLisa();

var hm1 = myPlot.Add.Heatmap(data);
hm1.Colormap = new ScottPlot.Colormaps.Turbo();

myPlot.Add.ColorBar(hm1);
```

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/a63eee17-7ac9-49dc-9f99-aab905097258)

I'll leave this issue open to track improving the layout.

**Comment by NateEbling** [comment]
Thanks for looking into this. From what I can tell, there is no way to add a ```Colorbar``` to the plot without first having data to generate a heatmap. Is that correct, or am I missing something? Previously with SP4.1, I simply used: 

```
myPlot.AddColorbar();
```

The idea is to generate a plot with all of the needed components before any data is supplied. The alternative seems like I'd have to add it at runtime once data is generated.

**Comment by swharden** [comment]
Hi @NateEbling, I think this has been fixed in #3610 and will be in the next release! 🚀 Thanks again for originally pointing this out.
